Project/DAQ_System/Program/Src/Platform/I2C/STM/I2CStmF4.cpp

```diff
@@ -0,0 +1,26 @@
+/*
+ * i2cHalWrapper_stmf4.cpp
+ *
+ *  Created on: Mar 4, 2024
+ *      Author: Nico
+ */
+#include <Src/Platform/I2C/STM/I2CStmF4.hpp>
+
+namespace platform {
+
+	I2CStmF4::I2CStmF4(I2C_HandleTypeDef &hi2c)
+			: _i2c_(hi2c) {
+	}
+
+	I2CStmF4::~I2CStmF4(){}
+
+	void I2CStmF4::Receive(uint8_t *response, uint8_t dev_add_, uint8_t response_size_){
+		HAL_I2C_Master_Receive(&_i2c_, dev_add_, response, response_size_, HAL_MAX_DELAY);
```

Comment:
Ditto.

Btw, if you're still interested in the I2C interface not requiring the `response_size_`, we can try this out: [vector.data()](https://cplusplus.com/reference/vector/vector/data/).

C++ vectors are heap-allocated, so they do not require a size to be initialized. They have a `.size()` method. But the interesting part is they also have `.data()`, which returns a raw-pointer to the data (which is essentially what `*response` is).

Reply:
Definitely wroth looking into, I have a list of things jotted down that we could look into while implementing the abstraction to the other sensors